src/x/sync/types.go

```diff
@@ -76,16 +66,32 @@ type PooledWorkerPool interface {
 	// iterations.
 	// This should only be used for code that can guarantee the wait time for a worker is low since if the ctx is not
 	// checked the calling goroutine blocks waiting for a worker.
-	FastContextCheck(batchSize int) PooledWorkerPool
+	FastContextCheck(batchSize int) StaticPooledWorkerPool
+}
+
+// DynamicPooledWorkerPool is like a StaticPooledWorkerPool, but can dynamically grow to ensure a client is never
+// blocked enqueuing work.
+type DynamicPooledWorkerPool interface {
+	// Init initializes the pool.
+	Init()
+
+	// AlwaysGo assigns the Work to be executed by a Goroutine, expanding the pool of goroutines to
+	// accommodate the work. The newly allocated goroutine will temporarily
+	// participate in the pool in an effort to amortize its allocation cost, but
+	// will eventually be killed. This allows the pool to dynamically respond to
+	// workloads without causing excessive memory pressure. The pool will grow in
+	// size when the workload exceeds its capacity and shrink back down to its
+	// original size if/when the burst subsides.
+	AlwaysGo(work Work)
```

Comment:
I chose a different method name (not `Go`), because I wanted to make sure a `StaticPooledWorkerPool` could not be used for a `DynamicPooledWorkerPool`.

Reply:
Makes sense, but nit here: could we make this `GoAlways(...)` instead of `AlwaysGo(..)`? That way at least it matches the other methods which all start with `Go...(...)` (like `GoWithTimeout`).